### PR TITLE
Feat/datetime

### PR DESCRIPTION
**Pull request by chrisrabe**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,23 +11,27 @@
   "dependencies": {
     "@apollo/client": "^3.3.21",
     "@expo-google-fonts/montserrat": "^0.2.0",
+    "@react-native-community/datetimepicker": "^3.5.2",
```
Comment (Tomas-Lawton) [inline package.json:14]: so cooooool

File: src/screens/Home/index.tsx [hunk 2]
```diff
@@ -10,12 +11,18 @@ interface HomeScreenProps {
 }
 
 const HomeScreen: React.FC<HomeScreenProps> = ({ appName }) => {
+  const [date, setDate] = useState<string>("2021-07-22T10:18:00.000Z");
```
Comment (winniezheng) [inline src/screens/Home/index.tsx:14]: default to today's date?

**Review by Tomas-Lawton — APPROVED** [review]

File: src/components/ui/DateTimeInput/DateTimeInput.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          <View>
+            <Icon id="date-picker" name="calendar" type="ionicon" />
+          </View>
+          <View style={styles.input}>
+            <Text style={styles.inputText}>{displayText}</Text>
+          </View>
+        </View>
+      </TouchableOpacity>
+      <View style={styles.errorContainer}>
+        {errorMessage && (
```
Comment (winniezheng) [inline src/components/ui/DateTimeInput/DateTimeInput.tsx:71]: should errorMessage be wrapped above the View error container component?

Comment (chrisrabe) [reply src/components/ui/DateTimeInput/DateTimeInput.tsx:71]: I was matching the styles of the react-native-elements input component. I think they still have the view there even without the error text

Comment (winniezheng) [reply src/components/ui/DateTimeInput/DateTimeInput.tsx:71]: hm but the View doesnt need to show unless there's an error message right?

File: src/components/ui/DateTimeInput/DateTimeInput.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      onConfirm(date.toUTCString());
+      setCurrentDate(date);
+      hideDatePicker();
+    },
+    [hideDatePicker, onConfirm],
+  );
+
+  const displayText = useMemo(() => {
+    if (currentDate) {
+      return dayjs(currentDate).format("DD MMMM [at] hh:mm a");
```
Comment (winniezheng) [inline src/components/ui/DateTimeInput/DateTimeInput.tsx:52]: not sure how dayjs works but does this format invalid string formats?

Comment (chrisrabe) [reply src/components/ui/DateTimeInput/DateTimeInput.tsx:52]: Good catch. it doesn't check for invalid dates atm. I just assume whatever is passed is a valid date string

**Pull request merged**